std/algorithm/comparison.d

```diff
@@ -1567,3 +1567,146 @@ unittest
                 2, "two",
                 ));
 }
+
+/** Returns: First Argument (element of $(D a)) whose implicit conversion to
+    $(D bool) is $(D true.
+
+    Similar to behaviour of or operator in dynamic languages such as Lisp's (or
+    a...) and Python's a or ....
+
+    TODO: Is inout Conversion!T the correct return value?
+
+    NOTE: Lazy parameters are currently inferred by DMD to alwasy throw. This
+    makes it impossible to mark $(D either) as $(D nothrow). Issue at
+    https://issues.dlang.org/show_bug.cgi?id=12647
+*/
+CommonType!Ts either(Ts...)(lazy Ts a)
+if (a.length >= 1 && !is(CommonType!Ts == void))
+{
+    foreach (e; a)
+        if (e)
+            return e;
+    return CommonType!Ts.init;
+}
+
+/// ditto
+auto ref either(Ts...)(ref Ts a)
+if (a.length >= 1 && allSameType!Ts)
+{
+    foreach (ref e; a[0 .. $ - 1])
+        if (e)
+            return e;
+    return a[$ - 1];
```

Comment:
It seems that the only sensible thing to do would be to throw an exception or assert, but that seems too heavy for such a small function. Maybe have the user provide a default value? It doesn't really make sense, but I don't see much of an alternative.